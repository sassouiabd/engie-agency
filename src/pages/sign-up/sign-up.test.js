import React from "react";
import { render, cleanup } from "@testing-library/react";
import SignUp from "./sign-up.page";
import * as reactRedux from "react-redux";

describe("signUp component", () => {
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  afterEach(cleanup);

  it("match snapshot reference", () => {
    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);

    expect(dummyDispatch).not.toHaveBeenCalled();

    const { asFragment } = render(<SignUp />);

    expect(asFragment(<SignUp />)).toMatchSnapshot();
  });
});
