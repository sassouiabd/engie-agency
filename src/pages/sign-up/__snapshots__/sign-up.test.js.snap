// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp component match snapshot reference 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-1"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Sign up
      </h1>
      <form
        class="makeStyles-form-3"
        novalidate=""
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="email"
                id="email-label"
              >
                Email Address
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Email Address *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Password *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign Up
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="/signin"
            >
              Already have an account? Sign in
            </a>
          </div>
        </div>
      </form>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright © 
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="https://www.engie.fr/"
        >
          engie-agency
        </a>
         2021.
      </p>
    </div>
  </main>
</DocumentFragment>
`;
